### ISSUE TITLE
Intermittent failures to read certificate from key vault

### ISSUE THREAD

**Issue by bammond** [hashicorp/terraform-provider-azurerm#22060]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.2

### AzureRM Provider Version

Appeared in 3.56.0, still present in 3.58.0

### Affected Resource(s)/Data Source(s)

data azurerm_key_vault_certificate

### Terraform Configuration Files

```hcl
data "azurerm_key_vault_certificate_data" "example_cert" {
  name         = "example"
  key_vault_id = data.azurerm_key_vault.key_vault.id
  version      = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
}
```

### Debug Output/Panic Output

```shell
│ Error: retrieving certificate "example" from keyvault: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="Operation get is not allowed on a disabled secret." InnerError={"code":"SecretDisabled"}
│ 
│   with module.istio-install[0].data.azurerm_key_vault_certificate_data.example_cert,
│   on istio_install/main.tf line 57, in data "azurerm_key_vault_certificate_data" "example_cert":
│   57: data "azurerm_key_vault_certificate_data" "example_cert" {
```

### Expected Behaviour

We are running this in our CI where we have a "prerequisites" job that creates the key vault, does the role assignment, and creates the certificate. This creates the certificate with "Enabled" state, but to be sure we also have a line that explicitly sets that to be true. Then we have a job that runs `terraform apply` on our code, and most of the time this works and successfully reads the certificate.

### Actual Behaviour

Approximately 1 in 10 runs of this we will hit the error above having just successfully pulled in the key vault as a data source

### Steps to Reproduce

1. Create a key vault
2. Assign yourself Certificates Officer role on the key vault
3. Create a secret
4. Run `terraform apply`

**Comment by wuxu92** [comment]
@bammond as you said `Appeared in 3.56.0`, does this mean the issue apears only after this version? and could you please share an example configuration of your certificate resource?

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. @wuxu92  was unable to reproduce this behavior. We’re going to close this issue for now, but if you run into this again with the latest Terraform and Provider versions, please feel free to [open a new issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) along with an example configuration of your certificate resource.
